app/src/main/java/io/github/doubi88/slideshowwallpaper/SlideshowWallpaperService.java

```diff
@@ -101,21 +101,18 @@ public SlideshowWallpaperEngine() {
         @Override
         public void onOffsetsChanged(float xOffset, float yOffset, float xOffsetStep, float yOffsetStep, int xPixelOffset, int yPixelOffset) {
             super.onOffsetsChanged(xOffset, yOffset, xOffsetStep, yOffsetStep, xPixelOffset, yPixelOffset);
-            float deltaXResult = 0;
-            if (currentImageHandler != null) {
-                ImageInfo currentImage = currentImageHandler.getCurrentImage();
-                if (currentImage != null) {
-                    Bitmap image = currentImage.getImage();
-                    if (image != null) {
-                        deltaXResult = calculateDeltaX(image, xOffset, xOffsetStep);
-                    } else {
-                        deltaXResult = 0;
-                    }
-                }
-            }
-            deltaX = deltaXResult;
+
+            this.xOffset = xOffset;
+            this.xOffsetStep = xOffsetStep;
             // When the xOffset is not a whole number, the wallpaper is scrolling
             isScrolling = (Math.floor(xOffset) != xOffset);
+
+            SharedPreferencesManager.TooWideImagesRule rule = manager.getTooWideImagesRule(getResources());
+            if (rule != SharedPreferencesManager.TooWideImagesRule.SCROLL_FORWARD && rule != SharedPreferencesManager.TooWideImagesRule.SCROLL_BACKWARD) {
+                // We don't need to retrigger drawing for scrolling if we don't have one of the scrolling settings enabled
+                return;
```

Comment:
Can you please invert the if statement and remove that return here, so the handler calls are inside the if? I know, that's a matter of taste, but I don't like return statements in the middle of a function.

Reply:
Done!